src/ui/Alert/Alert.tsx

```diff
@@ -0,0 +1,104 @@
+import { cva, VariantProps } from 'cva'
+import * as React from 'react'
+
+import { cn } from 'shared/utils/cn'
+
+import successImage from './assets/check-circle.svg'
+import warningImage from './assets/exclamation.svg'
+import infoImage from './assets/information-circle.svg'
+import errorImage from './assets/x-circle.svg'
+
+const alertVariants = cva(
+  '[&>svg]:text-foreground relative w-full border-l-4 p-4 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg~*]:pl-7',
+  {
+    variants: {
+      variant: {
+        error: 'border-ds-primary-red bg-error-100',
+        info: 'border-ds-blue-darker bg-ds-blue-nonary',
+        success: 'border-green-500 bg-green-100',
+        warning: 'border-orange-500 bg-orange-100',
+      },
+    },
+    defaultVariants: {
+      variant: 'info',
+    },
+  }
+)
+
+export function variantToIcon(variant?: string | null) {
+  let alt = 'information circle'
+  let src = infoImage
+  switch (variant) {
+    case 'error':
+      alt = 'error'
+      src = errorImage
+      break
+    case 'info':
+      alt = 'information circle'
+      src = infoImage
+      break
+    case 'success':
+      alt = 'checkmark in circle'
+      src = successImage
+      break
+    case 'warning':
+      alt = 'triangle exclamation'
+      src = warningImage
+      break
+  }
+  return (
+    <img className="float-left -mt-1 mr-2 align-middle" src={src} alt={alt} />
+  )
+}
+
+interface RootProps
+  extends React.HTMLAttributes<HTMLHeadingElement>,
+    VariantProps<typeof alertVariants> {}
+
+const AlertRoot = React.forwardRef<HTMLDivElement, RootProps>(
+  ({ className, variant, ...props }, ref) => (
+    <div
+      ref={ref}
+      role="alert"
+      className={cn(alertVariants({ variant }), className)}
+      {...props}
+    >
+      {variantToIcon(variant)}
```

Comment:
This child spacing issue is hard. I'm thinking ask design what they think of aligning the text like you have it here.

If not, you could solve it with wrapper sub-components (like in our Card):
- AlertHeader (contains the svg and top text, provide a font-weight variant)
- AlertBody (contains the bottom text)

Reply:
This feels like over-complication, but I can't think of a better solution atm. I think we just get design to agree to what you have lol

Reply:
Already have sign off from design on the current look :) But agree with you it's not the most elegant atm. 

I've thought about using grids, since the main "issue" is the icon is persistent regardless of the header or not. So being able to manipulate the image such that its always present, but can "push" the header text if that text is present is something that I'm sure can be done trivially if my CSS knowledge was a bit better 😂

Reply:
https://stackoverflow.com/questions/19179424/how-to-wrap-text-around-an-image-using-html-css

didn't know this was a thing. Could be possible for sure